### ISSUE TITLE
API mismatch for serializing keys

### ISSUE THREAD

**Issue by BSipos-RKF** [TimothyClaeys/pycose#32]
With the different implementations of CoseKey `encode` function, there are some discrepancies with how the attribute names are handled (adding or stripping leading underscore) and so the API breaks on some attr names. I'm not sure if I was using it correctly but I have made a patch which at least makes the uses consistent for `SymmetricKey`.

**Comment by TimothyClaeys** [comment]
Thanks again for the issue. I've updated the internal representation of the `CoseKey` class and its derivatives.

We can reassess when I release the new version on the develop branch.

**Comment by TimothyClaeys** [comment]
If you want you could open a PR with your patch. We can apply the fix until I'm done with my redesign of the code base.

**Comment by TimothyClaeys** [comment]
There were definitely some issues with the serialization of the COSE key objects. The API has been updated in #41.

COSE keys can now be constructed and serialized as follows.

Manually creating a COSE key:
```
from binascii import unhexlify

from cose.curves import P256
from cose.keys import CoseKey
from cose.keys.keyparam import KpKty, EC2KpCurve, EC2KpX, EC2KpY, KpKeyOps
from cose.keys.keytype import KtyEC2

cose_key = CoseKey.from_dict({
    KpKty: KtyEC2,
    EC2KpCurve: P256,
    EC2KpX: unhexlify(b'98F50A4FF6C05861C8860D13A638EA56C3F5AD7590BBFBF054E1C7B4D91D6280'),
    EC2KpY: unhexlify(b'F01400B089867804B8E9FC96C3932161F1934F4223069170D924B7E03BF822BB'),
    KpKeyOps: []
})

print(cose_key.encode())
```

Randomly created key:
```
from cose.keys import SymmetricKey

cose_key = SymmetricKey.generate_key(key_len=16)

print(cose_key.encode())
```

**Comment by TimothyClaeys** [comment]
Solved by #41 and #47
